### PR TITLE
fix(Scripts/TempestKeep): revamp Kael'Thas

### PR DESCRIPTION
**Pull request by Grimdhex**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [X] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

* Force AttackStop for all members of the threatlist before the transition event during the last
* Fix some unit flag logics
* Fix the reset logic
* Cleanup a little the script and remove unused enums

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/19548 and https://github.com/azerothcore/azerothcore-wotlk/issues/18486 . Partial fix for https://github.com/azerothcore/azerothcore-wotlk/issues/18488

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [X] This pull request requires further testing and may have edge cases to be tested.

* another test would be welcome. I found it difficult to reproduce the initial outcome. Even if the logic would dictate that it's fixed know.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue
- [X] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

- Enter Tempest Keep.
- Make your way to Kael'thas.
- Get to his final phase transition.
- Observe you can attack him despite not being able to attack him.

## Known Issues and TODO List:

- [ ] During the transition scene, when Kael'Thas descending he have a walk animation

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/TempestKeep/Eye/boss_kaelthas.cpp [hunk 1]
```diff
@@ -756,14 +734,39 @@ struct boss_kaelthas : public BossAI
             return;
 
         DoMeleeAttackIfReady();
+
+        ScheduleHealthCheckEvent(50, [&]{
```
Comment (Nyeriah) [inline src/server/scripts/Outland/TempestKeep/Eye/boss_kaelthas.cpp:738]: this will schedule a new check every milisecond

**Comment by Grimdhex** [comment]
Good for review again. Unfortunately OnDamageTaken was not working properly (possibility to go below 50% before than the transition scene was triggered). Samething for RemoveAllAttackers.

**Comment by alecsci** [comment]
@Grimdhex given your recent comment, can this PR be tested now ?

> Good for review again. Unfortunately the use of OnDamageTaken/HealthPctBelow was not working properly (possibility to go below 50% before than the transition scene was triggered). Samething for RemoveAllAttackers.

**Review by Nyeriah — APPROVED** [review]

**Comment by Grimdhex** [comment]
> @Grimdhex given your recent comment, can this PR be tested now ?
> 
> Also, do you re-worked this too [chromiecraft/chromiecraft#7074](https://github.com/chromiecraft/chromiecraft/issues/7074) ?

I didn't fix this issue

**Pull request merged**